ocaml/src/asd_server.ml

```diff
@@ -410,6 +410,109 @@ module Net_fd = struct
+       Some ("file_exists",ok) |> Lwt.return
+  in
+  let checksum_ok cv =
+    let (checksum, value) = cv in
+    let algo = Checksum.algo_of checksum in
+    Lwt.catch
+      (fun () ->
+        let hash = Hashes.make_hash algo in
+        match value with
+        | Value.Direct blob ->
+           let open Slice in
+           let () = hash # update_substring blob.buf blob.offset blob.length in
+           let checksum2 = hash # final () in
+           let ok = checksum2 = checksum in
+           let open Checksum.Algo in
+           Some (show algo, ok) |> Lwt.return
+        | Value.OnFs (fnr, size) ->
+           DirectoryInfo.with_blob_fd
+             dir_info fnr
+             (fun fd ->
+               Buffer_pool.with_buffer
+                 Buffer_pool.default_buffer_pool
+                 (fun buffer ->
+                   let reader = Lwt_bytes.read fd in
+                   let writer = hash # update_lwt_bytes_detached
+                   in
+                   Lwt_extra2.copy_using reader writer size buffer)
+             )
+           >>= fun () ->
+           let checksum2 = hash # final () in
+           let ok = checksum2 = checksum in
+           let open Checksum.Algo in
+           Some (show algo, ok) |> Lwt.return
+      )
+      (fun exn ->
+        let msg = Checksum.Algo.show algo ^ ":" ^ (Printexc.to_string exn) in
+        Some (msg , false) |> Lwt.return
+      )
+  in
+  let checks =
+    if verify_checksum
+    then
+      [exists_ok; checksum_ok]
+    else
+      [exists_ok]
+  in
+  perform_read
+    io_sched
+    prio
+    (fun () ->
+      let max = cap_max ~max () in
+      let (cnt, items), has_more =
+        Rocks_key_value_store.map_range
+          kv
+          ~first:(Keys.key_with_public_prefix first) ~finc
+          ~last:(match last with
+                 | None -> Keys.public_key_next_prefix
+                 | Some (last, linc) -> Some (Keys.key_with_public_prefix last, linc))
+          ~reverse
+          ~max
+          (fun cur key ->
+            Keys.string_chop_prefix key,
+            deserialize Value.from_buffer (Rocks_key_value_store.cur_get_value cur)
+          )
+      in
+      Lwt_list.fold_left_s
+        (fun (cnt, last_key, acc) (key, cv ) ->
+          Lwt_list.fold_left_s
+            (fun acc check ->
+              check cv >>= function
+              | None   -> Lwt.return acc
+              | Some r -> Lwt.return (r :: acc)
+            ) [] checks
+          >>= fun check_results ->
+          let acc' =
+            if show_all
+               || not (List.fold_left (fun  acc (_,ok) -> acc && ok ) true check_results)
+            then
+              (cnt+1, Some key, (key, check_results) :: acc)
+            else
+              (cnt, Some key, acc)
+          in
+          Lwt.return acc'
+        ) (0, None, []) items
+      >>= fun (cnt', last, r) ->
+      let r' = List.rev r in
+      let factor = if verify_checksum then 1000 else 100 in
```

Comment:
doubt it. for 16KB that's in rocksdb it's almost free. For 16KB+1B that's on the FS it's suddenly way more expensive. In all, it's continuous nor linear, and we're comparing apples and oranges anyway.